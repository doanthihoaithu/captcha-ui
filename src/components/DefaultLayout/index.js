import React, { Suspense } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import SideBar from 'components/SideBar'
import routes from 'routes';
// import { userServices } from 'services';

export default () => {
  return (
    <div className="home-container row overflow-hidden w-100 m-0 animated fadeIn fast ">
      <div className="col-1 pr-0 d-inline-block side-bar-root animated slideInLeft" style={{ "minHeight": "100vh" }}>
        <SideBar />
      </div>
      <div className="col-11 d-flex flex-column align-items-center">
        <Suspense fallback="Loading...">
          <Switch>
            {routes.appRoutes.map(route =>
              route.component ? (
                <Route
                  key={route.name}
                  path={route.path}
                  exact={route.exact}
                  name={route.name}
                  render={restProps => <route.component {...restProps} title={route.name} />}
                />
              ) : null,
            )}
            <Redirect from="/" to="/" />
          </Switch>
        </Suspense>
      </div>
    </div>
  );
};
