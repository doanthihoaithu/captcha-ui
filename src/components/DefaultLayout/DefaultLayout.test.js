import React from 'react'
import { shallow } from 'enzyme'

import DefaultLayout from '.'

const wrapper = shallow(
    <DefaultLayout />
)

it('button should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("div").hasClass('d-flex flex-column align-items-center')).toBe(true)
})