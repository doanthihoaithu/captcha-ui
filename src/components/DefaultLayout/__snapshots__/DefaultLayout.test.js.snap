// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should be rendered correctly 1`] = `
<div
  className="home-container row overflow-hidden w-100 m-0 animated fadeIn fast "
>
  <div
    className="col-1 pr-0 d-inline-block side-bar-root animated slideInLeft"
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <SideBar />
  </div>
  <div
    className="col-11 d-flex flex-column align-items-center"
  >
    <Suspense
      fallback="Loading..."
    >
      <Switch>
        <Route
          exact={true}
          key="Home"
          name="Home"
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          key="Change Password"
          name="Change Password"
          path="/change-password"
          render={[Function]}
        />
        <Redirect
          from="/"
          to="/"
        />
      </Switch>
    </Suspense>
  </div>
</div>
`;
